### PR TITLE
Fix obsolete api waring [#260] issue to support android project

### PR DESCRIPTION
**Pull request by KaelLuo**

### CODE REVIEW

**Comment by hal9002** [comment]
Can one of the admins review this PR?

**Comment by StefMa** [comment]
Is a task in the properties?
Why not checking if this task is available via the Task Container?

**Comment by KaelLuo** [comment]
Hey StefMa, thanks for the quick review.  
`variant` is not a task, it's actually passed from "ReleasePlugin#attachArtifacts" function within this code 
```
project.android.libraryVariants.all { variant ->
                String publicationName = variant.name
                MavenPublication publication = createPublication(publicationName, project, extension)
                new AndroidAttachments(publicationName, project, variant).attachTo(publication)
            }
```
In some other popular library, they also fix this kind of issue in same way, a similar fix discuss attached here:
https://github.com/google/protobuf-gradle-plugin/issues/295

File: plugin/core/src/main/groovy/com/novoda/gradle/release/internal/AndroidAttachments.groovy [hunk 1]
```diff
@@ -51,13 +51,22 @@ class AndroidAttachments extends MavenPublicationAttachments {
 
     private static Task androidJavadocsJarTask(Project project, String publicationName, def variant) {
         Javadoc javadoc = project.task("javadoc${publicationName.capitalize()}", type: Javadoc) { Javadoc javadoc ->
-            javadoc.source = variant.javaCompiler.source
-            javadoc.classpath = variant.javaCompiler.classpath
+            if(variant.hasProperty('javaCompileProvider')) {
+                javadoc.source = variant.javaCompileProvider.get().source
```
Comment (mr-archano) [inline plugin/core/src/main/groovy/com/novoda/gradle/release/internal/AndroidAttachments.groovy]: I would prefer to extract a little method where to put this if statement (like we have already done in [another plugin](https://github.com/novoda/gradle-static-analysis-plugin/blame/master/plugin/src/main/groovy/com/novoda/staticanalysis/internal/CodeQualityConfigurator.groovy#L133-L139)) so we can remove some duplication

**Review by mr-archano — CHANGES_REQUESTED** [review]
Thanks @KaelLuo for providing this. Just left a small remark.

**Comment by KaelLuo** [comment]
您收到此邮件表示您发送的邮件我已经收到，会尽快给您回复，请耐心等候，亲

**Pull request closed without merging**